nni/algorithms/compression/v2/pytorch/pruning/tools/base.py

```diff
@@ -440,3 +444,185 @@ def _compress_mask(self, mask: Tensor) -> Tensor:
+        save_dir = Path(self._log_dir_root, name)
+        model = torch.load(Path(save_dir, '{}_model.pth'.format(name)))
+        masks = torch.load(Path(save_dir, '{}_masks.pth'.format(name)))
+        with Path(save_dir, 'config_list.json').open('r') as f:
+            config_list = json_tricks.load(f)
+        return model, config_list, masks
+
+    def _save_origin_data(self, origin_model: Module, origin_config_list: List[Dict],
+                          origin_masks: Dict[str, Dict[str, Tensor]]):
+        self._save_data('origin', origin_model, origin_config_list, origin_masks)
+
+    def _load_origin_data(self) -> Tuple[Module, List[Dict], Dict[str, Dict[str, Tensor]]]:
+        return self._load_data('origin')
+
+    def _save_best_result(self, best_model: Module, best_config_list: List[Dict],
+                          best_masks: Dict[str, Dict[str, Tensor]]):
+        self._save_data('best', best_model, best_config_list, best_masks)
+
+    def _load_best_result(self) -> Tuple[Module, List[Dict], Dict[str, Dict[str, Tensor]]]:
+        return self._load_data('best')
+
+    def get_best_result(self) -> Optional[Tuple[int, Module, List[Dict], Dict[str, Dict[str, Tensor]], float]]:
+        """
+        Returns
+        -------
+         Optional[Tuple[int, Module, List[Dict], Dict[str, Dict[str, Tensor]], float]]
+            If self._best_task_id is not None,
+            return best task id, best pruned model, config list used in this task, masks on the pruned model, score.
+        """
+        if self._best_task_id is not None:
+            model, config_list, masks = self._load_best_result()
+            return self._best_task_id, model, config_list, masks, self._best_score
+        return None
+
+    def _save_task_result(self, task_id: int, model: Module, masks: Dict[str, Dict[str, Tensor]],
+                          score: Optional[float], up_model_masks: Dict[str, Dict[str, Tensor]]):
+        task = self._tasks[task_id]
+        if score is not None:
+            task.score = score
+            if self._best_score is None or score > self._best_score:
+                self._best_score = score
+                self._best_task_id = task_id
+                self._save_best_result(model, task.config_list, masks)
+        if self._save_result:
+            torch.save(model, Path(task.log_dir, 'pruned_model.pth'))
+            torch.save(masks, Path(task.log_dir, 'masks_on_pruned_model.pth'))
+            under_pruning_model, _, _ = self._load_task_data(task_id)
+            torch.save(under_pruning_model, Path(task.log_dir, 'under_pruning_model.pth'))
+            torch.save(up_model_masks, Path(task.log_dir, 'masks_on_under_pruning_model.pth'))
+
+    def _load_task_data(self, task_id: int) -> Tuple[Module, List[Dict], Dict[str, Dict[str, Tensor]]]:
+        task = self._tasks[task_id]
+        model = torch.load(task.model_path)
+        config_list = deepcopy(task.config_list)
+        masks = torch.load(task.masks_path)
```

Comment:
Is the default device behavior of `torch.load` reliable? Do we need to explicitly check device placement or move the model and masks to the same device?

Reply:
maybe we can add an interface in the future